src/Arcus.Security.Tests.Integration/HashiCorp/HashiCorpSecretProviderTests.cs

```diff
@@ -65,6 +64,172 @@ public async Task AuthenticateWithUserPassKeyValueV2_GetSecret_Succeeds()
             }
         }
 
+        [Fact]
+        public async Task AuthenticateWithUserPassKeyValueV2_GetNotFoundSecret_Fails()
+        {
+            // Arrange
+            string secretPath = "mysecret";
+            string secretName = "my-value";
+            string expected = "s3cr3t";
+
+            string userName = "arcus";
+            string password = "123";
+
+            using (HashiCorpVaultTestServer server = await StartServerWithUserPassAsync(userName, password, DefaultDevMountPoint))
+            {
+                await server.KeyValueV2.WriteSecretAsync(
+                    mountPoint: DefaultDevMountPoint,
+                    path: secretPath,
+                    data: new Dictionary<string, object> { ["unknown-prefix-" + secretName] = expected });
+
+                var authentication = new UserPassAuthMethodInfo(userName, password);
+                var settings = new VaultClientSettings(server.ListenAddress.ToString(), authentication);
+                var provider = new HashiCorpSecretProvider(settings, VaultKeyValueSecretEngineVersion.V2, DefaultDevMountPoint, secretPath);
+
+                // Act
+                string actual = await provider.GetRawSecretAsync(secretName);
+
+                // Assert
+                Assert.Null(actual);
+            }
+        }
+
+        [Fact]
+        public async Task AddHashiCorpVaultWithUserPass_WithMutationToRemovePrefix_Succeeds()
+        {
+            // Arrange
+            string secretPath = "secretpath";
+            string secretKey = "my-value", expected = "s3cr3t";
+            string userName = "arcus", password = "123";
+
+            var builder = new HostBuilder();
+
+            using (HashiCorpVaultTestServer server = await StartServerWithUserPassAsync(userName, password, DefaultDevMountPoint))
+            {
+                await server.KeyValueV2.WriteSecretAsync(
+                    mountPoint: DefaultDevMountPoint,
+                    path: secretPath,
+                    data: new Dictionary<string, object> { [secretKey] = expected });
+
+                // Act
+                builder.ConfigureSecretStore((config, stores) =>
+                {
+                    stores.AddHashiCorpVaultWithUserPass(
+                        server.ListenAddress.ToString(), userName, password, secretPath, keyValueMountPoint: DefaultDevMountPoint, 
+                        mutateSecretName: secretName => secretName.Remove(0, 5));
```

Comment:
Maybe use replace instead so that we can use a variable which is the same as [here](https://github.com/arcus-azure/arcus.security/pull/188/files#diff-7a06e78fdfe87689905c17dc86d12041R125)?
```suggestion
                        mutateSecretName: secretName => secretName.Replace(SecretNamePrefix, string.Empty));
```

Reply:
Hmm, well, it's just by accident that they share that value. Both test (should not) have anything in common as to say that they only can be used like that. I think it will make it less clear.

Reply:
But they are part of same test or what am I missing? 🤔 

We assume removing 0 to 5 is enough, but if we change "Test-" to something else it will break the test

Reply:
Yes, that's why it should stay local. Maybe I'll change the test so they both mutate something different to avoid the confustion.

Reply:
Ah, you mean local, you mean just a variable 🙄 . Ok, I can do that. 😄 .

Reply:
Yes, sorry!